test/AST-Quake/reset_after_measure.cpp

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: cudaq-quake %s | cudaq-opt --qubit-reset-before-reuse | FileCheck %s
+
+#include <cudaq.h>
+void no_reuse() __qpu__ {
+  cudaq::qubit q;
+  h(q);
+  mz(q);
+}
+
+// CHECK:   func.func @__nvqpp__mlirgen__function_no_reuse
```

Comment:
Let's use the generate-test-checks.py script (tpls/llvm/mlir/utils) to generate more comprehensive CHECK lines.